### ISSUE TITLE
Main Node is null when statically typing class variables with _ready function

### ISSUE THREAD

**Issue by zekrosem** [godotengine/godot#51856]
### Godot version

v4.0.dev.20210811.official [7188cb601]

### System information

Windows 10, Vulkan Mobile, GTX 3070 TI

### Issue description

Basically, when setting up the _ready method on a script. If one of the class vars is statically typed, then the node with the script attached is null. I've searched a bit in the issues list and found #51538 that might be related but I am not sure since it's yet quite different.

### Steps to reproduce

In the zip file below, you will find 2 scenes : 
- main_working.tscn
- main_bugged.tscn

The difference between these two scenes lies in the files : player.gd and player_working.gd at line 5.

Basically, I have just 
- created a main scene, 
- added a node 1 (CharacterBody2D) 
- attached a custom script to node 1 and defined it (class_name) as Player
- added a node 2 (CollisionShape2D) to node 1
- attached a custom script to node 2 and defined it (class_name) as PlayerCollisionShape
- added an @export var to node 1 
- added a variable with type PlayerCollisionShape to node 1

Removing the static type from last step gets rid of the issue.

### Minimal reproduction project

[reproduciblebug.zip](https://github.com/godotengine/godot/files/7009840/reproduciblebug.zip)

**Comment by dalexeev** [comment]
The bug is not reproducible in 4.0 RC 1.
